src/Controller/OcrController.php

```diff
@@ -262,6 +279,29 @@ public function apiAvailableLangsAction(): JsonResponse {
 		] );
 	}
 
+	/**
+	 * phpcs:disable MediaWiki.Commenting.FunctionAnnotations.UnrecognizedAnnotation
+	 * @Route("/api/available_line_ids", name="apiLineIds", methods={"GET"})
```

Comment:
I just noticed, the `?engine` query is required by the [`setup()` method](https://github.com/wikimedia/wikimedia-ocr/blob/79dc7d0598c33bc18932e78edcccea6e71d34028/src/Controller/OcrController.php#L91) to set the OCR engine for all further operations.

Reply:
Maybe it could set `static::$params['engine']` in this method? As it stands, people could pass e.g. `/api/available_line_ids?engine=google`, which wouldn't really make any sense.

Reply:
Yupp, doing that